app/lib/gate_keeper.rb

```diff
@@ -0,0 +1,9 @@
+class GateKeeper
+  def self.demo_environment?
+    if Rails.env.demo?
+      true
+    else
+      ENV["DEMO_MODE"] == "true"
```

Comment:
DEMO_MODE env variable exists so we can do acceptance on staging? But on the real demo app we'll set the Rails.env to be demo?

Reply:
Almost! I think we should have both staging and demo be set to DEMO_MODE=true. In the past we've had staging and demo both model the demo_environment behavior (banner, extra step alerting it's an example site, and not delivering real applications to the office).

I was sort of torn about the implementation of this method. Basically, we need an additional environment in the credentials.yml (for this, demo), but the demo behavior (described above) seems separate.

The other option I had considered was that this method should only key off of the demo_mode environment variable—and the demo environment is just for credentials. Now that I'm typing this out, I think that behavior is a bit more understandable, so I'd like to make that change.